src/election_anomaly/user_interface/__init__.py

```diff
@@ -730,16 +767,23 @@ def get_runtime_parameters(keys, param_file=None):
 	if len(p) == 0:
 		raise FileNotFoundError
 
-	for k in keys:
+	for k in required_keys:
 		try:
 			d[k] = parser['election_anomaly'][k]
 		except KeyError:
-			missing_params['missing'].append(k)
+			missing_required_params['missing'].append(k)
+
+	if optional_keys:
```

Comment:
You might be able to leverage the try/except block to catch both and reduce the amount of nesting. That may be somewhat of a stylistic issue, but also it's nice to minimize nesting where possible because it's easier to think/reason about.

Reply:
OK, reduced the nesting a different way. Couldn't see how to combine the two try/except blocks without making the logic more convoluted.